tentacle/src/transports/tcp_base_listen.rs

```diff
@@ -0,0 +1,488 @@
+                        Err(err) => {
+                            debug!("accept websocket stream err: {:?}", err);
+                        }
+                    },
+                }
+                return;
+            }
+            #[cfg(feature = "tls")]
+            UpgradeModeEnum::OnlyTLS => {
+                match crate::runtime::timeout(timeout, acceptor.accept(stream)).await {
+                    Err(_) => debug!("accept tls server stream timeout"),
+                    Ok(res) => match res {
+                        Ok(stream) => {
+                            let mut addr = socketaddr_to_multiaddr(remote_address);
+                            addr.push(Protocol::Tls(Cow::Borrowed("")));
+                            if sender
+                                .send((addr, MultiStream::Tls(Box::new(stream))))
+                                .await
+                                .is_err()
+                            {
+                                debug!("receiver closed unexpectedly")
+                            }
+                        }
+                        Err(err) => {
+                            debug!("accept tls server stream err: {:?}", err);
+                        }
+                    },
+                }
+                return;
+            }
+            #[cfg(feature = "tls")]
+            UpgradeModeEnum::TcpAndTLS => {
+                let mut peek_buf = [0u8; 16];
+                if let Err(e) = stream.peek(&mut peek_buf).await {
+                    debug!("stream encountered err: {}, close unexpectedly", e);
+                    return;
+                }
+
+                // The first sixteen bytes of secio's Propose message's mode is fixed
+                // it's bytes like follow:
+                //
+                // | 4 byte | 4 byte | 4 byte | 4 byte | 4 byte |...
+                // |--|--|--|--|--|
+                // | LengthDelimitedCodec header| molecule propose header | rand start | rand end/pubkey start | pubkey end/exchange start |...
+                //
+                // LengthDelimitedCodec header is big-end total len
+                // molecule propose header is little-end total len
+                // rand start offset is 24 = (5(feild count) + 1(total len))* 4
+                let length_delimited_header =
+                    u32::from_be_bytes(TryInto::<[u8; 4]>::try_into(&peek_buf[..4]).unwrap());
+                let molecule_header =
+                    u32::from_le_bytes(TryInto::<[u8; 4]>::try_into(&peek_buf[4..8]).unwrap());
+                let rand_start =
+                    u32::from_le_bytes(TryInto::<[u8; 4]>::try_into(&peek_buf[8..12]).unwrap());
+                let rand_end =
+                    u32::from_le_bytes(TryInto::<[u8; 4]>::try_into(&peek_buf[12..16]).unwrap());
+
+                if length_delimited_header == molecule_header
+                    && rand_start == 24
+                    && rand_start < rand_end
```

Comment:
Do we need to check the length of `rand_end - rand_start`?

Reply:
we can't, secio does not specify the size of rand